src/engines/clappy_trio.c

```diff
@@ -1,58 +1,249 @@
+        default:
+            sprite = 0xffff;
+            break;
+    }
+
+    anim = clappy_trio_get_anim(CLAPPY_TRIO_ANIM_CLAP);
+    
+    sprite_set_anim(gSpriteHandler, sprite, anim, 0, 1, 0x7f, 0);
+
+    volume = (gClappyTrio->lionClapVolume * 5) >> 3;
+    
+    play_sound_w_pitch_volume(&s_HC_seqData, volume, INT_TO_FIXED(2.0));
+}
+
+// Engine Event 04 (Set Manual Clap Volume)
+void clappy_trio_set_clap_volume(u32 volume) {
+    gClappyTrio->lionClapVolume = volume;
+}
+
+// Engine Event 03 (Enable Grayscale Effect)
+void clappy_trio_enable_grayscale(u8 enable) {
+    gClappyTrio->grayscale = enable;
+}
+
+// Game Engine Update
+void clappy_trio_engine_update(void) {
+    text_printer_update(gClappyTrio->textPrinter);
+}
+
+// Game Engine Stop (Stub)
+void clappy_trio_engine_stop(void) {
+}
+
+void clappy_trio_cue_spawn(struct Cue *cue, struct ClappyTrioCue *info, u32 smileAfter) {
+    info->smileAfter = smileAfter;
+}
+
+// Cue - Update
+u32 clappy_trio_cue_update(struct Cue *cue, struct ClappyTrioCue *data, u32 runningTime, u32 duration) {
+    if (runningTime > ticks_to_frames(0x78)) {
+        return TRUE;
+    }
+
+    return FALSE;
+}
+
+// Cue - Despawn (Stub)
+void clappy_trio_cue_despawn(void) {
+}
+
+// Cue - Hit (https://decomp.me/scratch/UAIPR)
+// This and clappy_trio_common_beat_animation match if clappy_trio_bg_pal 
+// is split into seperate variables, however doing so is bad practice
+// and should be attempted another way
 #include "asm/engines/clappy_trio/asm_080308f4.s"
 
-#include "asm/engines/clappy_trio/asm_080309a8.s"
-
-#include "asm/engines/clappy_trio/asm_080309f4.s"
-
-#include "asm/engines/clappy_trio/asm_08030a10.s"
-
+void clappy_trio_cue_barely(struct Cue *cue, struct ClappyTrioCue *info, u32 pressed, u32 released) {
+    struct Trio *trio = &gClappyTrio->trio;
+    struct Animation *clapAnim = clappy_trio_get_anim(CLAPPY_TRIO_ANIM_CLAP);
+
+    sprite_set_anim(gSpriteHandler, trio->sprites[3], clapAnim, 2, 1, 0x7F, 0);
+
+    play_sound(&s_tebyoushi_pati_seqData);
+
+    beatscript_enable_loops();
+}
+
+// Cue - Miss
+void clappy_trio_cue_miss(struct Cue *cue, struct ClappyTrioCue *info) {
+    struct Trio *trio = &gClappyTrio->trio;
+    trio->beatAnimation = CLAPPY_TRIO_ANIM_STATE_GLARE;
+    trio->resetBeatAnimation = 2;
+    beatscript_enable_loops();
+}
+
+// Input Event
+void clappy_trio_input_event(u32 pressed, u32 released) {
+    struct Trio *trio = &gClappyTrio->trio;
+    struct Animation *clapAnim = clappy_trio_get_anim(CLAPPY_TRIO_ANIM_CLAP);
+    
+    sprite_set_anim(gSpriteHandler, trio->sprites[3], clapAnim, 2, 1, 0x7F, 0);
+
+    play_sound(&s_witch_donats_seqData);
+    
+    trio->beatAnimation = CLAPPY_TRIO_ANIM_STATE_GLARE;
+    trio->resetBeatAnimation = 2;
+    
+    beatscript_enable_loops();
+}
+
+// Common Event 0 (Beat Animation) (https://decomp.me/scratch/UuWC8)
+// This matches, but also has issues with the palettes
 #include "asm/engines/clappy_trio/asm_08030a60.s"
```

Comment:
for nonmatchings, maybe also include a commented out version of the function to easily see what its doing? (make sure its functionally equivalent)